### DISCUSSION TITLE
how to create a new crud operation ( custom app) in saleor

### DISCUSSION THREAD

**Discussion by sruthi-pooliry** [saleor/saleor#13421] [Q&A]
Hi team ,
how can we create a new crud operation in a custom app 
where Iam:
Create a new custom table successfully using model, now getting error ; **ImproperlyConfigured: model is required for ModelMutation**

followed these steps to create custom app:
1. create app in saleor/saleor folder
2. cd <path>\saleor\saleor\
3. python <path>\saleor\manage.py startapp testapp
4. Define models in <path>\saleor\saleor\testapp
5. python D:\SaleOr\code\saleor\manage.py makemigrations
6. create app in graphQL
7. cd <path>\saleor\saleor\graphql\ 
8. python <path>\saleor\manage.py startapp testapp 
9. Created GraphQL schemas 

Where I want to be:
create new mutation for crud operation using the newly created model

Any help is appreciated!
<img width="745" alt="MicrosoftTeams-image (6)" src="https://github.com/saleor/saleor/assets/136053231/c17fd106-2935-4b42-b861-8b3394316ded">

**Comment by pranavsreepathi** [comment]
Hi @sruthi-pooliry, 

Have you checked whether you included the "model" option within the Meta class of your custom Mutation. This establishes the connection between the Model your Mutation is predominantly based upon and your Mutation. Perhaps study the ModelMutation class within saleor.graphql.core.mutations.py to further understand this.
